optics-core/src/Optics/Label.hs

```diff
@@ -290,6 +298,8 @@ import Optics.Internal.Optic
 -- prefixed, it works out:
 --
 -- @
+-- {-\# LANGUAGE DuplicateRecordFields #-}
```

Comment:
This isn't necessary here, right?

Reply:
I think you need at least `DisambiguateRecordFields` to allow you to use `name` instead of `User.Internal.name`, don't you? I debated which to add, but went for `DuplicateRecordFields` because it implies `DisambiguateRecordFields` and we're using it elsewhere. We could change this or add a comment with the alternative, though.

Reply:
You don't need it if only one `name` field is in scope. I just checked and you don't even need it if there are multiple `name` fields in scope, which makes sense as once you're inside record syntax for value construction (for update it's a different story I suppose), GHC knows what type you refer to so the choice is unambiguous.

Reply:
> which makes sense as once you're inside record syntax for value construction (for update it's a different story I suppose), GHC knows what type you refer to so the choice is unambiguous.

Right, this is precisely what `DisambiguateRecordFields` adds over `Haskell2010`: to be able to construct or pattern-match on a record with unqualified fields, if the fields are ambiguous or in scope only qualified. This makes sense because the data constructor tells you which datatype and hence which fields you mean. But it needs (at least) `DisambiguateRecordFields` enabled.

Reply:
Oh, you're right, it doesn't work without any extensions.

I was confused, because I tried it in a project and in worked without `DisambiguateRecordFields` nor `DuplicateRecordFields`. But the project enables `RecordWildCards` globally and it turns out it's enough :thinking:

Reply:
Yes, `RecordWildCards` also implies `DisambiguateRecordFields`. Really `DisambiguateRecordFields` should just be part of `GHC2020`... :wink: